tests/schema/test_info.py

```diff
@@ -62,3 +64,38 @@ def hello(self, info: Info[str, str]) -> Result:
         "variableValues": "{}",
         "returnType": "<class 'tests.schema.test_info.test_info_has_the_correct_shape.<locals>.Result'>",  # noqa
     }
+
+
+def test_return_type():
+    def resolver_factory(return_type, return_value):
+        def resolver(info) -> return_type:
+            # check return type
+            assert info.return_type == return_type
+            return return_value
+
+        return resolver
```

Comment:
could we use pytest parametrization for this? 😊

Reply:
Nice idea, but I'm not sure if it'd be easy task. I added different type of resolvers too. some return types are inherited from resolver, some from field

Reply:
we can split the test in two 😊

Reply:
that's even better idea =)

Reply:
Done